CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+## How do I contribute
+### Reporting Issues
+We use [GitHub Issues](https://github.com/ROCmSoftwarePlatform/MIOpen/issues) to track public **bugs** and **enhancement requests**.
+
+If you have found an issue, please check [MIOpen documentation](https://rocmsoftwareplatform.github.io/MIOpen/doc/html/) to see if it is a bug or new feature enhancement which is not supported in the latest version of MIOpen:
+
+#### Bugs
+Please follow the template below to report any bugs found in MIOpen:
+
+1. Description: ***Please be clear and descriptive***
+2. How to Reproduce the issue:
+* Hardware Information:
+* Docker environment or Software Version:
+* Expected behavior:
+* Actual behavior:
+3. Any additional information:
+
+#### Enhancement Requests
+Please follow the template below to report any enhancement requests for MIOpen:
+
+1. Description: ***Please be clear and descriptive***
+2. Value and Motivation:
+* Feature and functionalities enabled:
+* Any alternatives:
+3. Any additional information:
+
+The author must set labels (and assigns a miliestone) according to his/her own understanding.
+
+Other contributors can change these values if they disagree. That being said, 
+adding a small comment explaining the motivation is highly recommended. 
+In this way, we keep the process flexible while cultivating mutual understanding.
+
+[**Note**] Most likely, the labels like "bug", "feature" or "complexity*" 
+would not be changed. However, "value*" or "urgency*" might be from mutual 
+understanding.
+### Creating a Pull Request
+No changes are allowed to be directly committed to the develop 
+branch of the MIOpen repository. All authors are required to 
+develop their change sets on a separate branch and then create 
+a pull request (PR) to merge their changes into the develop branch.
+
+Once a PR has been created, a developer must choose two reviewers 
+to review the changes made. The first reviewer should be a 
+technical expert in the portion of the library that the changes 
+are being made in. You can find a list of these experts in 
+[MIOpen Issue #789](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/789)
+. The second reviewer should be a peer reviewer. This reviewer 
+can be any other MIOpen developer.
+
+## Responsibility of the Author
+The author of a PR is responsible for:
+ * Writing clear, well documented code
+ * Meeting expectations of code quality
+ * Verifying that the changes do not break current functionality
+ * Writing tests to ensure code coverage
+ * Report on the impact to performance
+
+## Responsibility of the Reviewer
+Each reviewer is responsible for verifying that the changes are 
+clearly written in keeping with the coding styles of the library, 
+are documented in a way that future developers will be able to 
+understand the intent of the added functionality, and will 
+maintain or improve the overall quality of the codebase.
+
+Reviewer's task checklist:
+1. Has the PR passed necessary CI?
```

Comment:
I want to clarify how we understand "passed necessary CI".
Is it only about tests in current CI which should be passed.
Or also to check that the coverage is sufficient and to ask the author to add tests to the CI, if necessary.

Reply:
@shurale-nkn "add tests to the CI" is indicated in Point 5 below, so passing CI here meant tests in current CI which should be passed only.

However, it's a good point about sufficient coverage, and we may have different set of CI for different types of PRs. How would someone new know about which set of CI needs to be passed? :(

Reply:
Right now we have a "full set" of tests. If "full set" is passed, then it is Ok.

It is also possible to manually narrow or extend the set or tests. This is also Ok provided that all the reviewers and the author agree on that.

Reply:
It may worth explaining this in contributing rules.

Reply:
Mostly my question is about unit tests and verification.
And 'full test' can't guarantee any of this.

Reply:
Maybe it is the contributor's job to always provide a subsection in the PR description about which subset of test he used for functionality regression. If this is generic changes, then it could be one of the `full test` stages. If else, a specific test stage need to be specified that can cover the changes.

Reply:
@shurale-nkn 
> Mostly my question is about unit tests and verification. And 'full test' can't guarantee any of this.

If so, then the "full test" has an issue and needs to be updated or fixed. BTW this is what you are doing right now with that "reshuffle and reduce test cases", right?

@junliume Yes. If the existing test set does not cover the PR, then the contributor and reviewers should take care of it. Ideally, add necessary tests in the same PR.

Reply:
>If so, then the "full test" has an issue and needs to be updated or fixed. BTW this is what you are doing right now with that "reshuffle and reduce test cases", right?

No, it's about our tests do testing only for fastest solution, not for all, so we can't guaranty correctness of all solvers.
I asked to add this feature long time ago. It should be written somewhere in the plans.

Reply:
>@junliume Yes. If the existing test set does not cover the PR, then the contributor and reviewers should take care of it. Ideally, add necessary tests in the same PR.

Or this